tests/Integration/ApiPlatform/PostCustomerTest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\ApiPlatform;
+
+use ApiPlatform\Symfony\Bundle\Test\ApiTestCase;
+use Tests\Resources\DatabaseDump;
+
+class PostCustomerTest extends ApiTestCase
+{
+    public static function setUpBeforeClass(): void
+    {
+        parent::setUpBeforeClass();
+        DatabaseDump::restoreTables(['group']);
+    }
+
+    public function testAddCustomerGroup(): void
+    {
+        $bearerToken = $this->getBearerToken();
+        static::createClient()->request('POST', '/new-api/customers/group', [
+            'auth_bearer' => $bearerToken,
+            'json' => [
+                'localizedNames' => [
+                    'test1',
+                    'test2',
+                ],
+                'reductionPercent' => [
```

Comment:
The purpose in this case is not to define each parameter of the decimal number The API client itself would provide a value like `reductionPercent => '10.000'` and this string value must then be injected into the related CQRS command BUT it needs to be converted into a `DecimalNumber` first

Ideally, this should be done in a "generic way" meaning not only work with `DecimalNumber` but with other simple objects like VOs So during the normalization (either handled by Symfony serializer, or by a custom conversion on our side, or maybe by adding a new normalizer in the serializer) we need to detect that the command has a parameter which is not scalar but is an object and we try a simple construction by providing the value from the request as the constructor parameter

Very simple idea of the expected solution based on reflection:

```php
if (is_object($parameterType)) {
    $commandParameterValue = new $parameterType($queryValue);
}
```

Reply:
I agree with @jolelievre .

If I understand correctly : if we accept this implementation, it means that for _every endpoint_ where we expect a DecimalNumber, we will require the API user to provide a payload with both parameters `number` and `exponent` right? Although `exponent` should always be 1 and is useless for end-user.

We need to find a way to change this. We need that, when the Serializer processes this payload, it is able to automatically convert the number to a DecimalNumber with exponent 1.

As a end user if I want to use the API to set a percent-value, I expect to send a payload with something like "10%" or "10" or "10.0" . Not `{number:10,exponent:1}`.

Reply:
It's possible to send the payload without exponent, like this :

`{
    "localizedNames": ["toto", "tata"],
    "reductionPercent": {"number": "0"},
    "displayPriceTaxExcluded": false,
    "showPrice": true,
    "shopIds": [1]
}`

WDYT ? @jolelievre @matks 
And if it's necessary we can add the exponent parameter.

Reply:
OK for me 👍 . If we don't burden the user with the `exponent` useless parameter, then it's good